From: Modestas Vainius <modestas@vainius.eu>
Subject: Add Debian package version to KAboutData

This patch adds Debian package version to the Help -> About dialog
and makes it visible in the bug report created with
"Help -> Report a bug".

Signed-off-by: Modestas Vainius <modestas@vainius.eu>

---
 src/App.cpp        |    9 ++++++++-
 src/CMakeLists.txt |    5 +++++
 2 files changed, 13 insertions(+), 1 deletions(-)

diff --git a/src/App.cpp b/src/App.cpp
index ccb5f74..f2ac927 100644
--- a/src/App.cpp
+++ b/src/App.cpp
@@ -88,9 +88,16 @@ int App::mainThreadId = 0;
 extern void setupEventHandler_mac(long);
 #endif
 
+#ifdef DEBIAN_PACKAGE_VERSION
+#  define _STRINGIFY(s) #s
+#  define _DEBIAN_APP_VERSION(debver) "(Debian " _STRINGIFY(debver) ")"
+#  define FULL_APP_VERSION APP_VERSION " " _DEBIAN_APP_VERSION(DEBIAN_PACKAGE_VERSION)
+#else
+#  define FULL_APP_VERSION APP_VERSION
+#endif
 
 AMAROK_EXPORT KAboutData aboutData( "amarok", 0,
-    ki18n( "Amarok" ), APP_VERSION,
+    ki18n( "Amarok" ), FULL_APP_VERSION,
     ki18n( "The audio player for KDE" ), KAboutData::License_GPL,
     ki18n( "(C) 2002-2003, Mark Kretschmann\n(C) 2003-2009, The Amarok Development Squad" ),
     ki18n( "IRC:\nirc.freenode.net - #amarok, #amarok.de, #amarok.es, #amarok.fr\n\nFeedback:\namarok@kde.org\n\n(Build Date: %1)" ).subs( __DATE__ ),
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 68eb7fb..bd9ec29 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -554,6 +554,11 @@ target_link_libraries(amaroklib
     mediadevicelib
 )
 
+if (DEBIAN_PACKAGE_VERSION)
+    MESSAGE(STATUS "Building Debian package version ${DEBIAN_PACKAGE_VERSION} ...")
+    set_source_files_properties(App.cpp PROPERTIES COMPILE_FLAGS -DDEBIAN_PACKAGE_VERSION=${DEBIAN_PACKAGE_VERSION})
+endif (DEBIAN_PACKAGE_VERSION)
+
 if(WIN32)
     target_link_libraries(amaroklib ${QT_QTWEBKIT_LIBRARY})
 endif(WIN32)
-- 
tg: (8d05d01..) debian/aboutdata_add_debian_version_feature (depends on: upstream)
